lamindb/_file.py

```diff
@@ -119,13 +120,25 @@ def process_data(
     elif isinstance(data, (pd.DataFrame, AnnData)):  # DataLike, spelled out
         storage = lamindb_setup.settings.storage.record
         memory_rep = data
+        if key is not None:
```

Comment:
Could this call the following functionality instead of implementing new logic?

https://github.com/laminlabs/lamindb/blob/51b220ddb09ff13d87b68b9c331d2cd7b60691c0/lamindb/dev/storage/file.py#L51-L65

Inferring the suffix correctly is a science on its own. :)

Reply:
I will check that.

Reply:
Done.